cmd/juju/action/call.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package action
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/charm.v6"
+	"gopkg.in/juju/names.v3"
+	"gopkg.in/yaml.v2"
+
+	"github.com/juju/juju/apiserver/params"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/juju/common"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/cmd/output"
+)
+
+// leaderSnippet is a regular expression for unit ID-like syntax that is used
+// to indicate the current leader for an application.
+const leaderSnippet = "(" + names.ApplicationSnippet + ")/leader"
+
+var validLeader = regexp.MustCompile("^" + leaderSnippet + "$")
+
+// nameRule describes the name format of an action or keyName must match to be valid.
+var nameRule = charm.GetActionNameRule()
+
+func NewCallCommand() cmd.Command {
+	return modelcmd.Wrap(&callCommand{})
+}
+
+// runCommand enqueues an Action for running on the given unit with given
+// params
+type callCommand struct {
+	ActionCommandBase
+	api           APIClient
+	unitReceivers []string
+	leaders       map[string]string
+	actionName    string
+	paramsYAML    cmd.FileVar
+	parseStrings  bool
+	wait          waitFlag
+	out           cmd.Output
+	args          [][]string
+}
+
+const callDoc = `
+Queue an Action for execution on a given unit, with a given set of params.
+The Action ID is returned for use with 'juju show-action-output <ID>' or
+'juju show-action-status <ID>'.
+
+Valid unit identifiers are: 
+  a standard unit ID, such as mysql/0 or;
+  leader syntax of the form <application>/leader, such as mysql/leader.
+
+If the leader syntax is used, the leader unit for the application will be
+resolved before the action is enqueued.
+
+Params are validated according to the charm for the unit's application.  The
+valid params can be seen using "juju actions <application> --schema".
+Params may be in a yaml file which is passed with the --params option, or they
+may be specified by a key.key.key...=value format (see examples below.)
+
+Params given in the CLI invocation will be parsed as YAML unless the
+--string-args option is set.  This can be helpful for values such as 'y', which
+is a boolean true in YAML.
+
+If --params is passed, along with key.key...=value explicit arguments, the
+explicit arguments will override the parameter file.
+
+Examples:
+
+    juju call mysql/3 backup --wait
+    juju call mysql/3 backup
+    juju call mysql/leader backup
+    juju show-action-output <ID>
```

Comment:
I am not sure that you need 'show-action-output example among 'call' command examples :D

Reply:
show-action-output is still a valid command to see the result of calling an action.
It will be renamed however to show-operation in the next PR